libvmaf/include/libvmaf/model.h

```diff
@@ -70,6 +70,22 @@ typedef struct VmafModelCollectionScore {
     } bootstrap;
 } VmafModelCollectionScore;
 
+typedef struct VmafBuiltInModel {
+    const char *version;
+    const char *data;
+    const int *data_len;
+} VmafBuiltInModel;
```

Comment:
Can you call this struct `VmafModelDescription`? Let's just keep `version` in there for now, there should be no need to expose `data` or `data_len`.

Reply:
I believe that exposing the entire model is necessary for the pointer math in vmaf_model_next to work
I could be wrong. It's been a long time since I've written c++ so I'm pretty rusty. If we expose vmaf_model_descriptor with just the version field publicly but use vmafbuiltinmodel internally would the pointer math still work?

Reply:
Just for clarification, vmaf_model_next works on a pointer to a vmafbuiltinmodel struct with 3 pointers. In order to get the next item in the array, it needs to know how wide a vmafbuiltinmodel is, so if we only wanted to expose the version pointer in a struct in the header would we do something like

model.c
```c
typedef struct VmafBuiltInModel{
const char *version;
const char *data;
const int *data_len;
} VmafBuiltInModel;

const VmafBuiltInModel *vmaf_model_descriptor_next(const VmafBuiltInModel *prev){ etc...}

```

and in model.h
```c
typedef struct VmafModelDescriptor{
const char *version;
}

const VmafModelDescriptor *vmaf_descriptor_next(const VmafModelDescriptor *prev);
```

so within model.c vmaf_model_descriptor_next is doing pointer math on the VmafBuiltInModel struct while code calling this function assumes it's using VmafModelDescriptor?

Reply:
actually it occurs to me now that I can define vmafmodeldescriptor in model.h and keep the function signature of vmaf_model_descriptor_next but within the actual function body cast that pointer into a pointer to vmafbuiltinmodel then we should be good?

Reply:
Assuming my pointer math is right and c lays out the memory of properties in a struct in the order they're declared, this should be fixed with the latest commits